app/scripts/metamask-controller.js

```diff
@@ -6709,6 +6709,99 @@ export default class MetamaskController extends EventEmitter {
     );
   }
 
+  /**
+   * Creates middleware hooks that are shared between the Eip1193 and Multichain engines.
+   *
+   * @private
+   * @param {string} origin - The connection's origin string.
+   * @returns {object} The shared hooks.
+   */
+  setupCommonMiddlewareHooks(origin) {
```

Comment:
thoughts on regrouping these by type and bringing back the comments describing each group? like `// network configuration-related`

Reply:
good call! lmk if I got it right 🙏